ClassicsKotlin/app/src/main/java/com/android/tv/classics/receivers/TvLauncherReceiver.kt

```diff
@@ -104,20 +103,21 @@ class TvLauncherReceiver : BroadcastReceiver() {
      * Helper method used to retrieve a metadata item object given its program ID. Restricted API
      * is being used to compared programs based on their ID.
      */
-    @SuppressLint("RestrictedApi")
     private fun getMediaItemFromProgramId(
             context: Context, db: TvMediaDatabase, programId: Long?): TvMediaMetadata? {
 
         // Early exit: program ID is null
         if (programId == null) return null
 
         // Retrieve the program that matches this ID
-        val programItem =
-                TvLauncherUtils.getPreviewPrograms(context).find { it.id == programId } ?:
-                TvLauncherUtils.getWatchNextPrograms(context).find { it.id == programId }
-
+        val previewProgramItem =
+                TvLauncherUtils.getPreviewPrograms(context).find { it.id == programId }
+        // Retrieve the corresponding metadata item and return
+        if(previewProgramItem != null)
+            return previewProgramItem.let { db.metadata().findById(it.contentId) }
         // Retrieve the corresponding metadata item and return
-        return programItem?.let { db.metadata().findById(it.contentId) }
+        val watchNextProgramItem = TvLauncherUtils.getWatchNextPrograms(context).find { it.id == programId }
+        return watchNextProgramItem?.let { db.metadata().findById(it.contentId) }
```

Comment:
I don't see how this more verbose code block is an improvement over the inline assignment using an elvis operator with less code repetition.

Reply:
@ittiam-sham Is there any benefit to expanding the existing code block?